lib/src/features/line_renderer.dart

```diff
@@ -61,7 +69,48 @@ class LineRenderer extends FeatureRenderer {
         continue;
       }
 
-      context.canvas.drawPath(line, effectivePaint);
+      // do we need a dashed line?
+      if (style.lineLayout!.dashArray.length >= 2) {
+        final dashedLine = dashPath(line, CircularIntervalList(dashLengths));
+        context.canvas.drawPath(dashedLine, effectivePaint);
+      } else {
+        context.canvas.drawPath(line, effectivePaint);
+      }
+    }
+  }
+
+  // convert a path into a dashed path with given intervals
+  Path dashPath(Path source, CircularIntervalList<num> dashArray) {
+    final Path dest = Path();
+    for (final PathMetric metric in source.computeMetrics()) {
+      // start point of dashing
+      double distance = .0;
+      bool draw = true;
+      while (distance < metric.length) {
+        final num len = dashArray.next;
+        if (draw) {
+          dest.addPath(metric.extractPath(distance, distance + len), Offset.zero);
+        }
+        distance += len;
+        draw = !draw;
+      }
+    }
+
+    return dest;
+  }
+}
+
+// Fixed list always rotating through elements
+class CircularIntervalList<T> {
```

Comment:
Can we move this out of the line renderer? Does it need to be generic or can we make it specific to numbers?
It's not a list, so how about something like `RingNumberSupplier` or similar?